Source/OCMock/OCMockObject.m

```diff
@@ -31,15 +31,80 @@
 #import "OCMExpectationRecorder.h"
 #import "OCMQuantifier.h"
 
+@class XCTestCase;
+
+// gCurrentMocksToStopRecorder will point to either gTestCaseMocksToStop or gTestSuiteMocksToStop
+// depending on whether we are currently recoring testcase mocks or test suite mocks.
+// Controlled by OCMockXCTestObserver.
+static NSHashTable<OCMockObject *> *gTestCaseMocksToStop;
+static NSHashTable<OCMockObject *> *gTestSuiteMocksToStop;
+static NSHashTable<OCMockObject *> *gCurrentMocksToStopRecorder;
+
+// Flag that controls whether we should be asserting after stopmocking is called.
+// Controlled by OCMockXCTestObserver.
+static BOOL gAssertOnCallsAfterStopMocking;
+
+// "Fake" Protocol so we can avoid having to link to XCTest, but not get warnings about
+// methods not being declared.
+@protocol OCMockXCTestObservation
++ (id)sharedTestObservationCenter;
+- (void)addTestObserver:(id)observer;
+@end
 
-@implementation OCMockObject
+@interface OCMockXCTestObserver : NSObject
+@end
 
+@implementation OCMockObject
 #pragma mark  Class initialisation
++ (void)load
+{
+    gTestCaseMocksToStop = [[NSHashTable weakObjectsHashTable] retain];
+    gTestSuiteMocksToStop = [[NSHashTable weakObjectsHashTable] retain];
+    gCurrentMocksToStopRecorder = nil;
+    gAssertOnCallsAfterStopMocking = YES;
```

Comment:
The need for this is definitely a smell. I understand the problem, but I'm worried about the complexity that can arise from this additional logic, like "raise when the mock was stopped, but don't raise when it was stopped in a certain way". This'll make diagnosing issues (and answering questions on StackOverflow!) even harder.